### ISSUE TITLE
Email verification and reset password email links don't work

### ISSUE THREAD

**Issue by markvital** [hagopj13/node-express-boilerplate#172]
I'm trying to test user creation and reset password workflow.
It seems that email verification link in the email and reset password link don't work.

When I create user via api _/auth/register_ and then invoke _/auth/send-verification-email_ I got the following email:
<img width="741" alt="Screen Shot 2022-01-24 at 9 48 25 AM" src="https://user-images.githubusercontent.com/742796/150735718-c52eeb78-7e22-4b51-8330-faa4ee229d95.png">

The link is not clickable and it should point to local instead (or any other hostname provide in configuration).

Same for reset password, when I invoke _/auth/forgot-password_ api call with specified email address I got the following email:

<img width="746" alt="Screen Shot 2022-01-24 at 9 49 15 AM" src="https://user-images.githubusercontent.com/742796/150735905-d7781382-3cdf-4318-8f71-8073e7be4c6e.png">

**Comment by markvital** [comment]
Closely looking at the code I noticed that there is no routes to ```http://localhost/verifyEmail``` and ```http://localhost/reset-password```.
There are only api routes for correspondent POST queries (```/auth/verifyEmail``` and ```/auth/reset-password```) but it should be routes for GET queries (```/verifyEmail?token=XYZ``` and ```_reset-password?email=user@example.com```), so the can be invoked by clicking link in the email.

Or am I missing something?

**Comment by vmundhra** [comment]
@markvital yes you are missing the point. These two implementations expect you to have a front-end application running with page routes "/verify-email" and "/reset-password" respectively. The {link-to-app} is the domain of your front-end app. So lets assume you have a React SPA app running on localhost:3000 and this boilerplate running on localhost:8080. In the boilerplate code you would change link-to-app to localhost:3000. i.e. So when the email goes out the link in the email will be http://localhost:3000/verify-email?token=eyJhb....7w0. So when you click on the link a browser tab will open up that link and load your react app with the /verify-email page. Additionally, in your react app code for that page, you have to extract the token query param value and make an api call to /auth/verifyEmail with token. This now will verify the email of user. If the api returns success, you show some sort of verification complete message in UI. Similarly for reset-password have a page with route /reset-password, extract the token and keep it, but dont make the api call automatically. This page will have a form input with new password. when user enters new password and clicks submit, then make the call to /auth/reset-password with the token and new password. On success of api,  show some success msg. 

If you dont have an UI, you can test the api using postman. After getting the link on email, open up postman and call the api with value of the token. If the service returns success, check db that the email is verified to confirm.

Needless to say, when you deploy you server and front-end in production, the {link-to-app} will be the domain of the app and not localhost. e.g. https://app.acme.com/reset-password.

Hope you get it now.
